includes/import-functions.php

```diff
@@ -132,58 +131,66 @@ function edd_import_coupon_csv_file() {
 						$validation_error = true;
 						$validation_msg  .= __('Discount Name is blank.', 'edd') . "; ";
 					}
-					
+
 					// discount code  blank
 					if ( 0 == strlen( $edd_discount_code ) ) {
 						$validation_error = true;
 						$validation_msg  .= __('Discount code is blank.', 'edd') . "; ";
 					}
-					
+
+					// Verify only accepted characters.
+					$sanitized_discount_code = preg_replace( '/[^a-zA-Z0-9-_]+/', '', $edd_discount_code );
+					if ( strtoupper( $edd_discount_code ) !== strtoupper( $sanitized_discount_code ) ) {
+						$validation_error = true;
+						$validation_msg  .= __('Discount code is invalid.', 'edd') . "; ";
+					}
+
 					// discount type is blank
 					if ( 0 == strlen( $edd_discount_type ) ) {
 						$validation_error = true;
 						$validation_msg  .= __('Discount type is blank.', 'edd') . "; ";
 					}
-					
-					// discount type is invaid
-					if ( ( $edd_discount_type != 'flat' ) && ( $edd_discount_type != 'percentage' ) ) {
+
+					// discount type is invalid
+					$discount_types = array( 'flat', 'percentage' );
+					if ( ! in_array( $edd_discount_type, $discount_types, true ) ) {
 						$validation_error = true;
 						$validation_msg  .= __('Discount type is invalid.', 'edd') . "; ";
 					}
-					
+
 					// discount code is blank
 					if ( 0 == strlen( $edd_discount_code ) ) {
 						$validation_error = true;
 						$validation_msg  .= __('Discount code is blank.', 'edd') . "; ";
 					}
-					
+
 					// start time validation
 					if ( ! strtotime( $edd_discount_start ) ) {
 						$validation_error = true;
 						$validation_msg  .= sprintf(__('Date validation failed for string: %s', 'edd'), $edd_discount_start) . "; ";
 					}
-					
+
 					// end time validation
 					if ( ! strtotime( $edd_discount_end ) ) {
 						$validation_error = true;
 						$validation_msg  .= sprintf(__('Date validation failed for string: %s', 'edd'), $edd_discount_end) . "; ";
 					}
-					
-					// set start time to correct date type
-					$edd_discount_start =  date( 'm/d/Y H:i:s', strtotime( $edd_discount_start ) );
-					
-					// set end time to the correct date type
-					$edd_discount_end =  date( 'm/d/Y H:i:s',  strtotime(  date( 'm/d/Y', strtotime( $edd_discount_end ) ) . ' 23:59:59' )  );
-				
-					// check status 
+
+					// check status
 					$status = array( 'active', 'inactive', 'expired' );
-					if ( ! in_array( $edd_discount_status, $status ) ) {
+					if ( ! in_array( $edd_discount_status, $status, true ) ) {
 						$validation_error = true;
 						$validation_msg  .= __('Status needs to be active, inactive, or pending', 'edd') . "; ";
 					}
-					
-					// maybe more validation later
 
+					// Check product condition.
+					$product_conditions = array( 'all', 'any' );
+					if ( ! in_array( $edd_discount_product_condition, $product_conditions, true ) ) {
```

Comment:
I think this maybe needs to be changed. The sample spreadsheet has `0` for the product condition and so the import is failing with this check. One option would be to allow `0` as a value:
```suggestion
					if ( ! empty( $edd_discount_product_condition ) && ! in_array( $edd_discount_product_condition, $product_conditions ) ) {
```
However, I'm concerned that doing that would break if a user expects `1` to be a valid value. I don't think it should be, though, and it's not in the docs at all, so maybe just adding the empty check should be enough?

Reply:
To be honest, I'm not even sure what `0` or `1` would even mean :) since they are clearly not supported in either EDD 3.0 or EDD 2.10 (I didn't test older versions).  That's why I asked at the beginning:

> In example CSV file, values for some arguments are `0` or `1` (so `false` or `true`). That works for `not_global` and `use_once`. But it's not working for `product_condition` because it accepts only strings: `all` or `any`. ([In `EDD_CLI::create_discounts()`](https://github.com/easydigitaldownloads/easy-digital-downloads/blob/289400c2f83d792c6ea645ef9eb2962fae1455a1/includes/class-edd-cli.php#L777-L797), `0` and `1` are meant to be array indexes; should we use that functionality or switch to strings?)

I believe that this was either bug in example CSV all the time, or that EDD changed the way it supports this at some point. I can easily commit your suggestion but I am not sure should we do it without adding clear support what `0` or `1` mean.

Reply:
I'd rather keep `all` / `any` and update the sample accordingly.